pycontrails/core/aircraft_performance.py

```diff
@@ -543,3 +586,54 @@ class AircraftPerformanceGridData(Generic[ArrayOrFloat]):
 
     #: Engine efficiency, [:math:`0-1`]
     engine_efficiency: ArrayOrFloat
+
+
+def _fill_low_altitude_with_isa_temperature(vector: GeoVectorDataset, met_level_max: float) -> None:
+    """Fill low-altitude NaN values in ``air_temperature`` with ISA values.
+
+    The ``air_temperature`` param is assumed to have been computed by
+    interpolating against a gridded air temperature field that did not
+    necessarily extend to the surface. This function fills points below the
+    lowest altitude in the gridded data with ISA temperature values.
+
+    This function operates in-place and modifies the ``air_temperature`` field.
+
+    Parameters
+    ----------
+    vector : GeoVectorDataset
+        GeoVectorDataset instance associated with the ``air_temperature`` data.
+    met_level_max : float
+        The maximum level in the met data, [:math:`hPa`].
+    """
+    air_temperature = vector["air_temperature"]
+    is_nan = np.isnan(air_temperature)
+    low_alt = vector.level > met_level_max
+    cond = is_nan & low_alt
+
+    t_isa = vector.T_isa()
+    air_temperature[cond] = t_isa[cond]
```

Comment:
You could come up with slightly more reasonable temperature profiles by using the ISA lapse rate but matching temperatures at the lowest level of the met data. That would make the implementation significantly more complicated (you'd need some information about met data temperatures in this function), and I'm not sure how much it would actually improve fuel burn estimates.... So I don't have a strong opinion about whether this is worth your time.

Reply:
I thought about doing something like this. Without having a good source of ground truth (for example, ERA5 data all the way down to the surface), I decided not to pursue anything more sophisticated.